src/python/pants/backend/python/dependency_inference/import_parser.py

```diff
@@ -35,40 +34,55 @@ def all_imports(self) -> FrozenOrderedSet[str]:
         return FrozenOrderedSet(sorted([*self.explicit_imports, *self.inferred_imports]))
 
 
-def parse_file(*, filename: str, content: str) -> Optional[Tuple]:
+_Visitor = TypeVar("_Visitor")
+
+
+class VisitorInterface:
```

Comment:
This is being used as a protocol for type hints, right? If so, it'd be good to subclass `Protocol` to make that intent clear: https://mypy.readthedocs.io/en/stable/protocols.html

(Good idea to use a protocol!)

Reply:
If you use a protocol, I think you won't need to subclass this anymore down below.

Reply:
Yes! Done!